client/app/components/compose.coffee

```diff
@@ -464,13 +483,27 @@ ComposeEditor = React.createClass
                         document.execCommand('delete', false, null)
                         node.focus()
 
+
+            # Webkit/Blink and Gecko have some different behavior on
+            # contentEditable, so we need to test the rendering engine
+            # Webklink doesn't support insertBrOnReturn, so this will return
+            # false
+            gecko = document.queryCommandEnabled 'insertBrOnReturn'
+
             # Some DOM manipulation when replying inside the message.
             # When inserting a new line, we must close all blockquotes,
             # insert a blank line and then open again blockquotes
             jQuery('.rt-editor').on('keypress', (e) ->
-                if e.keyCode is 13
-                    # timeout to let the editor perform its own stuff
-                    setTimeout ->
+                if e.keyCode isnt 13
+                    # yes, our styleguide returning before the end of a
+                    # function. But they also don't like lines longer than
+                    # 80 characters
```

Comment:
Subfunctions are the right response to this problem. Try to stick to guidelines, it really helps others.